core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -340,11 +340,12 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
                 }
             }
 
+            Set<String> sensitiveVars = build.getSensitiveBuildVariables();
+            final VariableResolver<String> resolver = new Union<String>(new ByMap<String>(env), vr);
+            args.addKeyValuePairsFromPropertyString("-D", this.properties, resolver, sensitiveVars);
```

Comment:
@olivergondza @MarkEWaite Do you mean 2.19.3? I'm fine with it once the tests get finalized